tests/api/utils/__init__.py

```diff
@@ -0,0 +1,13 @@
+from typing import Dict, Iterable, List, Callable
+
+
+def get_messages_by_predicate(
+    messages: Iterable[Dict], predicate: Callable[[Dict], bool]
+) -> List[Dict]:
+    return [msg for msg in messages if predicate(msg)]
+
+
+def get_messages_by_keys(messages: Iterable[Dict], **keys) -> List[Dict]:
```

Comment:
What is this function doing ? Can you add a docstring to explain why we need it ?